### PR TITLE
chore(issue_template): replace `bug.md` with `bug.yml` issue form

### PR DESCRIPTION
**Pull request by Fdawgs**
See https://github.com/fastify/fastify/discussions/3051

#### Checklist

- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Can you add a screenshot of what it would look like?

**Comment by Fdawgs** [comment]
> 
> 
> Can you add a screenshot of what it would look like?

Can't seem to. 😞 
Merged this PR into my fork and i can't even use the issue form (I assume because I am not in the issue form beta).

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  - type: checkboxes
+    id: prerequisites
+    attributes:
+      label: Prerequisites
+      options:
+      - label: I have written a descriptive issue title
+        required: true
+      - label: | 
+          I have searched [existing issues](https://github.com/fastify/fastify/issues)
```
[the remark this answers is not in the document]
Comment (Fdawgs) [reply .github/ISSUE_TEMPLATE/bug.yml]: Have removed the link entirely. Unable to direct it to the current repo it seems.

**Comment by mcollina** [comment]
You can create a fresh repo in this org and test there

**Comment by Fdawgs** [comment]
image

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Is this ready for review?

**Comment by Fdawgs** [comment]
> 
> 
> Is this ready for review?

Yep!

**Comment by mcollina** [comment]
Let's see if @jsumners has any objection and then we can land.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    id: steps-to-reproduce
+    attributes:
+      label: Steps to Reproduce
+      description: |
+        List of steps, sample code or a link to code or a project that reproduces the behavior.
+        Make sure you place a stack trace inside a code (```) block to avoid linking unrelated issues.
```
Comment (jsumners) [inline .github/ISSUE_TEMPLATE/bug.yml:71]: Add a link here to the GitHub formatting documentation. I can say with extreme confidence that there is a vast population of developers who do not understand the triple backticks without some education.

**Review by jsumners — APPROVED** [review]
LGTM with a tweak.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 2 above)
Comment (Fdawgs) [reply .github/ISSUE_TEMPLATE/bug.yml]: Added in https://github.com/fastify/.github/pull/8/commits/0aaac973e9169e901a039f74087eee3d2c6e5350!

**Comment by mcollina** [comment]
Do not issue the updates for all repos, let's see what this gives us

**Comment by Fdawgs** [comment]
@mcollina, this repo provides the issue templates for all public Fastify repos, see the [community file documentation here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file).

If you just want it in the main Fastify repo for now, this commit will need to be reverted, and i'll open up a new PR just in the Fastify repo.

**Pull request merged**